### PR TITLE
Fix 3 bugs found by fuzzing

### PR DESCRIPTION
**Pull request by mcarton**
This PR fixes 3 bugs that have been found by `cargo fuzz`.

### CODE REVIEW

**Comment by mcarton** [comment]
I signed it!

File: src/parse.rs [hunk 1]
```diff
@@ -1549,7 +1549,11 @@ impl<'a> RawParser<'a> {
         while i < data.len() {
             match data.as_bytes()[i] {
                 b'>' => return i + 1,
-                b'\n' => i += self.scan_whitespace_inline(&data[i..]),
+                b'\n' =>{
```
Comment (Thomasdezeeuw) [inline src/parse.rs]: forgot a space after `=>`

**Comment by mcarton** [comment]
Fixed.

**Comment by raphlinus** [comment]
Thanks, appreciate the fixes!

**Pull request merged**